org.eclipse.scanning.api/src/org/eclipse/scanning/api/malcolm/attributes/TableAttribute.java

```diff
@@ -43,4 +43,31 @@ public void setValue(MalcolmTable tableValue) {
 	public void setHeadings(String[] headings) {
 		this.headings = headings;
 	}
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = super.hashCode();
+		result = prime * result + Arrays.hashCode(headings);
+		result = prime * result + ((tableValue == null) ? 0 : tableValue.hashCode());
+		return result;
+	}
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (!super.equals(obj))
+			return false;
+		if (getClass() != obj.getClass())
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') NullPointerException might be thrown as 'obj' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2259)